### PR TITLE
Add XL-friendly layout.

### PR DESCRIPTION
**Pull request by Sesamestrong**
## Description

This includes centering content Medium-style and keeping the footer always at least at the bottom of the page. See #207 for information on exactly what this implements.
Uses flexbox and ```margin:0px auto;``` to center things. Uses ```max-width``` to set the maximum width of the main content. This creates an effect similar to what Medium uses. The main content is set to be ```min-height:100vh;```, and is structured with Flexbox around a) a ```div``` holding the main content and b) the ```footer```. The ```div``` has property ```flex-grow:1;```, allowing it to grow to satisfy the ```min-height:100vh;``` requirement. This places the footer aligned at the bottom.

I used ```@media``` queries to limit width of items only when the viewport width is greater than ```1280px```. The max content width limit under XL circumstances is ```1140px```.

## Related Issue

#207

### ```1244px``` viewport width (Original site, my changes):

Original site, width <1280px
Changed site, same width

### ```>1280px``` viewport width (Original site, my changes):

Original site, width >1280px

Changed site, same width

### CODE REVIEW

**Comment by Sesamestrong** [comment]
Will do!

**Comment by Sesamestrong** [comment]
I have added a change. Here is the new behavior (only changed on XL screens):

image

**Comment by samajammin** [comment]
Hey @Sesamestrong thanks for continuing work on this. 

I'm seeing this on the preview homepage on my laptop (1440px width):

deploy-preview-624--ethereumorg netlify com_

Looks like one of your changes is now forcing this content to align left. Same with the /build page.

Please resolve that.

**Comment by Sesamestrong** [comment]
I actually have a question about this. The reason I aligned it left was so that there could be space-efficient division between content and the sidebar. Should I keep it left-aligned when a sidebar is present, or should I center the main content in all cases and just limit the max-width in order to make it not overlap with the sidebar?

**Comment by Sesamestrong** [comment]
Option 1: Maintain centering, reduce width when sidebar is present

image

2. Remove centering when a sidebar is present

image

**Comment by Sesamestrong** [comment]
I went with option 2.

**Comment by Sesamestrong** [comment]
I believe that it has all features that you want now.

File: docs/.vuepress/theme/components/Sidebar.vue [hunk 1]
```diff
@@ -154,4 +154,8 @@ function resolveOpenGroupIndex (route, items) {
     .sidebar
       display block !important
       transform translateX(0)
+
+@media only screen and (min-width:1280px)
```
Comment (samajammin) [inline docs/.vuepress/theme/components/Sidebar.vue]: We should add this width as a variable in config.styl

Comment (samajammin) [reply docs/.vuepress/theme/components/Sidebar.vue]: We do: `$breakL`

File: docs/.vuepress/theme/Layout.vue [hunk 2]
```diff
@@ -177,5 +181,24 @@ export default {
...
+
+header
+  margin 0px auto
+
+@media only screen and (min-width:$breakL)
+  #formatter,header
+    max-width 1140px
+  #wrapper.has-sidebar main
+    margin-left 10px
+    max-width "calc(1140px - %s - 10em)" % $sidebarWidth
```
Comment (samajammin) [inline docs/.vuepress/theme/Layout.vue]: We should add this width as a variable in config.styl

**Comment by Sesamestrong** [comment]
@samajammin I have made a commit on my repo with the changes that you have requested. Should I make a new PR or is it possible to re-merge this one?

**Comment by samajammin** [comment]
Thanks, @Sesamestrong.

I've reverted this PR in https://github.com/ethereum/ethereum-org-website/pull/634 - go ahead & make a new PR with all your changes (and please make the PR against the `dev` branch, not `master`). You can learn more about our deployment process here: https://github.com/ethereum/ethereum-org-website#deployment-lifecycle

**Pull request merged**